packages/aws-rfdk/lib/lambdas/nodejs/lib/configure-spot-event-plugin/test/spot-event-plugin-client.test.ts

```diff
@@ -238,4 +256,208 @@ describe('SpotEventPluginClient', () => {
+
+    // WHEN
+    await spotEventPluginClient.addGroups(addedGroupsCollection);
+
+    // THEN
+    const requestsCount = addedGroupsCollection.length > 0 ? 1 : 0;
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].GetRequest).toBeCalledTimes(requestsCount);
+
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].PostRequest).toBeCalledTimes(requestsCount);
+    if (requestsCount>0) {
+      // eslint-disable-next-line dot-notation, jest/no-conditional-expect
+      expect(spotEventPluginClient['deadlineClient'].PostRequest).toBeCalledWith(
+        '/db/settings/collections/groups/save',
+        {
+          Pools: Array.from(new Set(currentGroupsCollection.concat(addedGroupsCollection))),
+          ObsoletePools: obsoletePools,
+        },
+        {
+          headers: { 'Content-Type': 'application/json; charset=utf-8' },
+        },
+      );
+    }
+  });
+
+  test.each([
+    [ [], ['pool1', 'pool2'] ],
+    [['pool1', 'pool2'],['pool1', 'pool2']],
+    [['pool1', 'pool2'],['pool1', 'pool3']],
+    [['pool1', 'pool2'],[]],
+  ])('successful call addPool with existing pools %s and added pools %s', async (currentPoolsCollection: string[], addedPoolsCollection: string[]) => {
+    // GIVEN
+    const obsoletePools = ['obsolete_pool'];
+    const poolResponse: Response = {
+      data: {
+        Pools: currentPoolsCollection,
+        ObsoletePools: obsoletePools,
+      },
+      fullResponse: new IncomingMessage(new Socket()),
+    };
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].GetRequest = jest.fn().mockResolvedValue(poolResponse);
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].PostRequest = jest.fn().mockReturnValue(true);
+
+    // WHEN
+    await spotEventPluginClient.addPools(addedPoolsCollection);
+
+    // THEN
+    const requestsCount = addedPoolsCollection.length > 0 ? 1 : 0;
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].GetRequest).toBeCalledTimes(requestsCount);
+
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].PostRequest).toBeCalledTimes(requestsCount);
+    if (requestsCount>0) {
+      // eslint-disable-next-line dot-notation, jest/no-conditional-expect
+      expect(spotEventPluginClient['deadlineClient'].PostRequest).toBeCalledWith(
+        '/db/settings/collections/pools/save',
+        {
+          Pools: Array.from(new Set(currentPoolsCollection.concat(addedPoolsCollection))),
+          ObsoletePools: obsoletePools,
+        },
+        {
+          headers: { 'Content-Type': 'application/json; charset=utf-8' },
+        },
+      );
+    }
+  });
+
+  test.each([
+    [poolsColection.Pools,1],
+    [[], 0],
+  ])('successful call addPool with %s', async (poolsCollection: string[], requestsCount: number) => {
```

Comment:
This test looks like it covers the same code path that the above test we're adding in is covering. Should we remove this test?